packages/core/src/node/backend-application-module.ts

```diff
@@ -85,4 +86,7 @@ export const backendApplicationModule = new ContainerModule(bind => {
 
     bind(WsRequestValidator).toSelf().inSingletonScope();
     bindContributionProvider(bind, WsRequestValidatorContribution);
+
+    bind(WsLifecycleNotifier).toSelf().inSingletonScope();
+    bindContributionProvider(bind, WsLifecycleNotificationContribution);
```

Comment:
The logic here was to allow a reference to a specific socket to be retained in a map; such a reference would allow the websocket connection to the terminated due to an external event (JWT expiry, for example). Certainly open to advice on how to make this better!

Reply:
Then I think this patch is good! I was wondering if we'd ever need such events, and here we are :)

I'd be more comfortable with the names I proposed, this way we can later add callbacks for other events coming from the messaging system.

Reply:
More than happy to change the name. Let me take a look!

Reply:
@marechal-p I've made the changes locally but I'd just like your confirmation - we'll have "MessagingListener" and "WsRequestValidator" when I add these changes; do we definitely want the two different naming conventions?

Reply:
My reasoning here is that `WsRequestValidator` is a bad API name since it's a bit too specific. After your change, I might deprecate `WsRequestValidator` and move its bits into that `MessagingListener`. When we need to listen for more events from the messaging system, we will be able to just add methods to the interface of those contributions.

Reply:
Ah, I did wonder about that as well but I didn't want to make too much of a change as my first commit! :-)